include/util/transaction.h

```diff
@@ -12,9 +12,21 @@ namespace util {
     template <typename Rollback>
     class transaction {
     public:
-        transaction(Rollback rollback):
+        explicit transaction(Rollback rollback):
             _rollback{ std::forward<Rollback>(rollback) } {}
 
+        /**
+         *  Copy and move constructors
+         */
+        transaction(const transaction &other) = default;
+        transaction(transaction &&other) noexcept = default;
```

Comment:
Since Rollback is a template, can we guarantee this is noexcept? I'd say it should be _conditionally noexcept_. You wouldn't need to do anything for it, since `default`ed function have this as their default behaviour.

Reply:
Fixed